src/tree/SubscriptionTreeItem.ts

```diff
@@ -100,28 +106,26 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
 
         const promptSteps: AzureWizardPromptStep<IAppServiceWizardContext>[] = [];
         const executeSteps: AzureWizardExecuteStep<IAppServiceWizardContext>[] = [];
+
         promptSteps.push(new SiteNameStep());
-        promptSteps.push(new FunctionAppStackStep());
 
         const storageAccountCreateOptions: INewStorageAccountDefaults = {
             kind: StorageAccountKind.Storage,
             performance: StorageAccountPerformance.Standard,
             replication: StorageAccountReplication.LRS
         };
 
-        if (version === FuncVersion.v1) { // v1 doesn't support linux
-            wizardContext.newSiteOS = WebsiteOS.windows;
-        }
-
-        if (!context.advancedCreation) {
+        if (!wizardContext.advancedCreation) {
             LocationListStep.addStep(wizardContext, promptSteps);
             wizardContext.useConsumptionPlan = true;
-            wizardContext.stackFilter = getRootFunctionsWorkerRuntime(language);
+            wizardContext.stackFilter = getRootFunctionsWorkerRuntime(wizardContext.language);
             executeSteps.push(new ResourceGroupCreateStep());
-            executeSteps.push(new AppServicePlanCreateStep());
             executeSteps.push(new StorageAccountCreateStep(storageAccountCreateOptions));
-            executeSteps.push(new LogAnalyticsCreateStep());
             executeSteps.push(new AppInsightsCreateStep());
+            if (!context.dockerfilePath) {
+                executeSteps.push(new AppServicePlanCreateStep());
+                executeSteps.push(new LogAnalyticsCreateStep());
```

Comment:
This is used for the `AppInsightsCreateStep` to make it an workspace enabled AI. I think it should stay in the common steps.

Reply:
Hmm ok. I originally put this here since the CA api created the log so if I am to add this back then we would need to change the api

Reply:
We should probably verify if there are duplicate steps and remove them if there are. This might have to be an azuretools change though.

Reply:
After discussion offline is seems like the App Insights just checks to see if a log analytics exists in the same region and if not it creates one.

Reply:
I'll go ahead and approve this, but I'll end up consolidating everything into one command as discussed offline.